In [2]:
import requests
import os
from bs4 import BeautifulSoup
import re
req=requests.get('https://www.sanfoundry.com/')
soup=BeautifulSoup(req.text)
all_links=soup.find('div',class_='entry-content').find_all('a')
d={}
for i in all_links:
    d[i.text]=i.get('href')
# 452 keys
# 1 to 10, 11 to 20,21 to 30
start=int(input('From Key:'))
stop=int(input('To Key:'))
sid=int(input('Folder Number:'))
print(f'Scrapping from {start} to {stop}.')
stop+=1
key=start
for x,subject in enumerate(list(d.keys())[start:stop]):
    if re.match('1000.*?MCQs',subject):
        name_sub="_".join(subject.split()[1:-1])
        link_sub=d[subject]
        r_sub=requests.get(link_sub)
        soup_sub=BeautifulSoup(r_sub.text)
        sub=soup_sub.find('div',class_='entry-content')
        print('-------------------------------------------------------------------------------------------')
        print(f'--> Subject Key -->',key)
        print(f'--> Writing -->',subject)
        print('-------------------------------------------------------------------------------------------')
        for topic,table in zip(sub.find_all('h4'),sub.find_all('table')):
            print('\t---------------------------------------------------------------------------------------')
            print("\t--> Writing -->",topic.text)
            print('\t---------------------------------------------------------------------------------------')
            t_name='-'.join(re.split(';|,|\*|\n|/|\\|\.|:','_'.join(topic.text.split())))
            try:
                os.makedirs(f"SF\\{sid}. {name_sub}\\{t_name}\\Questions")
                os.makedirs(f"SF\\{sid}. {name_sub}\\{t_name}\\Answers")
            except OSError:
                pass
            for st in table.find_all('a'):
                questions="----Project By Karan Agrawal & Chirag Agarwal----\n\n\n"
                answers="----Project By Karan Agrawal & Chirag Agarwal----\n\n\n"
                sub_topic=st.text
                print("\t\t--> Writing -->",sub_topic)
                req_st=requests.get(st.get('href'))
                soup_st=BeautifulSoup(req_st.text)
                for que in soup_st.find('div',class_='entry-content').find_all('p')[1:-3]:
                    questions+="\n"+que.text.split('View Answer')[0]+"\n"
                    if que.find_next_sibling() and que.find_next_sibling().name=='pre':
                        questions+=que.find_next_sibling().text+'\n'
                    if que.find('img') and que.find('img').get('src'):
                        questions+=que.find('img').get('src')+'\n'
                    if que.find_next_sibling() and que.find_next_sibling().get('class'):
                        if que.find_next_sibling().get('class')[0]=='hk1_style-wrap5' :
                            for li in que.find_next_sibling().find_all('li'):
                                questions+=li.text+'\n'
                        elif que.find_next_sibling().get('class')[0]=="sf-mobile-ads" :
                            for li in que.find_next_sibling().find_next_sibling().find_all('li'):
                                questions+=li.text+'\n'
                for ans in soup_st.find('div',class_='entry-content').find_all('div',class_='collapseomatic_content'):
                    answers+="\n"+ans.text+"\n"
                f_name='-'.join(re.split(';|,|\*|\n|/|\\|\.|:',('_'.join(sub_topic.split()))))
                try:
                    with open(f"SF\\{sid}. {name_sub}\\{t_name}\\Questions\\{f_name}.txt",'w',encoding='utf-8') as fp:
                        questions+="\n\n----Project By Karan Agrawal and Chirag Agarwal----\n"
                        fp.write(questions)
                    with open(f"SF\\{sid}. {name_sub}\\{t_name}\\Answers\\{f_name}.txt",'w',encoding='utf-8') as fp:
                        answers+="\n\n----Project By Karan Agrawal and Chirag Agarwal----\n"
                        fp.write(answers)
                except FileNotFoundError:
                    print('File Not Found Error Handled')
        sid+=1
    key+=1
        
    print('-------------------------------------------------------------------------------------------')

From Key:300
To Key:414
Folder Number:138
Scrapping from 300 to 414.
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-----------

	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Mold and Mold Making  
	---------------------------------------------------------------------------------------
		--> Writing --> Types of Molds
		--> Writing --> Molding Processes
		--> Writing --> Methods of Ramming Mold Boxes – 1
		--> Writing --> Methods of Ramming Mold Boxes – 2
		--> Writing --> Methods of Green Sand Molding
		--> Writing --> Hand Molding Equipment
		--> Writing --> Steps Involved In Making a Mold
		--> Writing --> Venting of Moulds
		--> Writing --> Forces Acting on Cores and Mold
		--> Writing --> Mold-Metal Reaction
	---------------------------------------------------------------------------------------
	--> Writing -->  6. Questions on Pattern and Mould, Principles of Gating & Risering 
	---------------------------------------------------------------------------------------
		--> Writing --> Green Sand Mould
		--> Writing --> Fu

		--> Writing --> Tem Tech Thermal Sand Reclamation
		--> Writing --> Testing of Core Sands
	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Metal Furnace and Practice 
	---------------------------------------------------------------------------------------
		--> Writing --> Selection of Remelting Furnace
		--> Writing --> Cupola
		--> Writing --> Zones and Effiicency of Cupola
		--> Writing --> Coke Bed Charge Calculations
		--> Writing --> Air Requirements and Metallurgical Control for Cupola
		--> Writing -->  Cupola Charge Calculations
		--> Writing --> Reverberatory Furnace
		--> Writing --> Rotory Melting Furnace
		--> Writing --> Open Hearth Surface
		--> Writing --> Converter
		--> Writing --> Crucibles – Advantages and Production
		--> Writing --> Crucible Furnaces
		--> Writing --> Electric Furnace
		--> Writing --> Direct Arc Furnace
		--> Writing --> Rocking Furnace
		--> Writing --> Inducti

		--> Writing --> Lathe : Taper and Taper Turning
		--> Writing --> Methods of Taper Turning-1
		--> Writing --> Methods of Taper Turning-2
		--> Writing --> Lathe : Thread Cutting
		--> Writing --> Lathe : Calculation of Change Gears
		--> Writing --> Lathe : Cutting Parameters
		--> Writing --> Lathe : Numericals
	---------------------------------------------------------------------------------------
	--> Writing -->  3. Questions & Answers on Drilling Machine  
	---------------------------------------------------------------------------------------
		--> Writing --> Taper Shank Twist Drill Machine
		--> Writing --> Table Top Drilling Machine
		--> Writing --> Pillar Drilling Machine
		--> Writing --> Column Drilling Machine
		--> Writing --> Radial Drilling Machine
		--> Writing --> Hand Drilling and Gang Drilling
		--> Writing --> Turret Type Drilling Machine
		--> Writing --> Multi Spindle Drilling Machine
		--> Writing --> Micro-Mini Drilling Machine
		--> Writing --> Deep Hole D

		--> Writing --> IBM-Introduction, Material Removal Rate
		--> Writing --> IBM-Surface effects and Applications
	---------------------------------------------------------------------------------------
	--> Writing -->  6. Questions on Hybrid Electrochemical Processes 
	---------------------------------------------------------------------------------------
		--> Writing --> ECG-Introduction
		--> Writing --> ECG-MRR
		--> Writing --> ECG-Applications, Pros and Cons
		--> Writing --> Electrochemical Honning-Introduction
		--> Writing --> ECH-Process Characteristics, Applications
		--> Writing --> Electrochemical Superfinishing
		--> Writing --> Electrochemical Buffing
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
--> Subject Key --> 368
--> Writing --> 1000 IC Engine MCQs
--------------------------------------------------------------------------------

		--> Writing --> Effects of Supercharging
		--> Writing --> Turbocharging
	---------------------------------------------------------------------------------------
	--> Writing -->  16. Questions on Two Stroke Engines 
	---------------------------------------------------------------------------------------
		--> Writing --> Terminologies and Definations
		--> Writing --> Scavenging Process
		--> Writing --> Two-Stroke Engine
		--> Writing --> Scavenging Pumps
		--> Writing --> Advantages of Two Stroke Engines
		--> Writing --> Disadvantages of Two Stroke Engines
		--> Writing -->  Comparision Of Two Stroke S.I. and C.I. Engines – 1
		--> Writing --> Comparision of Two Stroke S.I. and C.I. Engines – 2
		--> Writing --> Advantages of Four Stroke Engines
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
--> Subject Key --> 369
--> Writing --> 1000 Power Pla

		--> Writing --> Sources of Energy – I
		--> Writing --> Miscellaneous
		--> Writing --> Environmental Degradation & Use of Renewable Energy-I
		--> Writing --> Environmental Degradation & Use of Renewable Energy-II
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
--> Subject Key --> 370
--> Writing --> 1000 Steam and Gas Turbines MCQs
-------------------------------------------------------------------------------------------
	---------------------------------------------------------------------------------------
	--> Writing -->  1. Questions & Answers on Basics and Classification 
	---------------------------------------------------------------------------------------
		--> Writing --> Types
		--> Writing --> Flow Passages and Arrangement Classification
		--> Writing --> Types, Engine and Component – 1
		--> Writing --> Types, Engine and Component – 

		--> Writing --> Parts of an Automobile
		--> Writing --> Description of Automobiles
		--> Writing --> Performance of an Automobile
	---------------------------------------------------------------------------------------
	--> Writing -->  2. Questions on Internal Combustion Engines 
	---------------------------------------------------------------------------------------
		--> Writing --> Heat Engines
		--> Writing --> Development of IC Engines
		--> Writing --> Applications of IC Engines
		--> Writing --> Energy Cycle
		--> Writing --> Parts of an IC Engine Cylinder and Liner
		--> Writing --> Various Parts of Engine – 1
		--> Writing --> Various Parts of Engine – 2
		--> Writing --> IC Engine Crankshaft and Camshaft
		--> Writing --> Engine Valves and their Mechanisms
		--> Writing --> Fuel Injection Systems
		--> Writing --> Flywheel
		--> Writing --> 2 Stroke Engines
	---------------------------------------------------------------------------------------
	--> Writing -->  3. Questi

		--> Writing --> Engine Starting Methods – 2
		--> Writing --> Lubrication System, Intake and Exhaust Systems
	---------------------------------------------------------------------------------------
	--> Writing -->  4. Questions on Hydro Electric Power Plant 
	---------------------------------------------------------------------------------------
		--> Writing --> Hydrology – 1
		--> Writing --> Hydrology – 2
		--> Writing --> Rainfall and its Measurement – 1
		--> Writing --> Rainfall and its Measurement – 2
		--> Writing --> Hydrographs – 1
		--> Writing --> Hydrographs – 2
		--> Writing --> Site Selection for Hydro Electric Power Plants
		--> Writing -->  Classification of Hydro-Plant
	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Nuclear Power Plant 
	---------------------------------------------------------------------------------------
		--> Writing --> Atomic Structure
		--> Writing --> Nucle

		--> Writing --> Global Warming 
		--> Writing --> Climate Change
		--> Writing --> Pollution Control Boards and Control Pollution Acts in India
		--> Writing --> Nuclear Hazards and Accidents
		--> Writing --> Need for Public Awareness and Institutions in Environment
	---------------------------------------------------------------------------------------
	--> Writing -->  9. Questions & Answers on Background to International Efforts for Environmental Protection
	---------------------------------------------------------------------------------------
		--> Writing --> International Environmental Centers
	---------------------------------------------------------------------------------------
	--> Writing -->  10. Questions on Human Population and the Environment  
	---------------------------------------------------------------------------------------
		--> Writing --> Population Growth 
		--> Writing --> Human Rights 
		--> Writing --> HIV and AIDS
		--> Writing --> Environment and Hum

		--> Writing --> Statistical Process Control for Short Production Runs – 6
		--> Writing --> Control Charts for Multiple-Stream Processes – 1
		--> Writing --> Control Charts for Multiple-Stream Processes – 2
		--> Writing --> Control Charts for Multiple-Stream Processes – 3
	---------------------------------------------------------------------------------------
	--> Writing -->  10. Questions on Basic Experimental Design and Designed Experiments 
	---------------------------------------------------------------------------------------
		--> Writing --> What is Experimental Design – 1
		--> Writing --> What is Experimental Design – 2
	---------------------------------------------------------------------------------------
	--> Writing -->  11. Questions & Answers on Basic Acceptance Sampling Procedures 
	---------------------------------------------------------------------------------------
		--> Writing --> Acceptance Sampling Introduction – 1
		--> Writing --> Acceptance Sampling Intr

		--> Writing --> Gear Measurement
	---------------------------------------------------------------------------------------
	--> Writing -->  14. Questions on Machine Tool Metrology 
	---------------------------------------------------------------------------------------
		--> Writing --> Machine Tools Tests
		--> Writing --> Alignment Tests on Lathe
		--> Writing --> Alignment Tests on Milling Machine
		--> Writing --> Alignment Tests on Pillar Type Drilling Machine
	---------------------------------------------------------------------------------------
	--> Writing -->  15. Questions & Answers on Measuring Machines & Total Quality Management 
	---------------------------------------------------------------------------------------
		--> Writing --> Coordinate Measuring Machines
		--> Writing --> Three Dimensional Measuring Machine
		--> Writing --> Universal Measuring Machine
		--> Writing --> Numerical Controlled Coordinate Measuring Machines
		--> Writing --> Computers in Metrology


		--> Writing --> Ceramics – 1
		--> Writing --> Ceramics – 2
		--> Writing --> Thermoplastics – 1
		--> Writing --> Thermoplastics – 2
		--> Writing --> Thermosetting Plastics, Particle Reinforced Composites
	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Composite & Ceramic Materials 
	---------------------------------------------------------------------------------------
		--> Writing --> Fibre Reinforced Composites & Other Composites
		--> Writing --> Protective Coatings
		--> Writing --> Ceramic Formation & Clay Ceramics
		--> Writing --> Abrasives, Silicate Structures and Polymorphism
	---------------------------------------------------------------------------------------
	--> Writing -->  6. Questions on Refractories & Dielectric Materials 
	---------------------------------------------------------------------------------------
		--> Writing --> Properties of Refractories
		--> Writing --> Refra

		--> Writing --> Point Imperfections
		--> Writing --> Line Imperfections 
		--> Writing --> Surface Imperfections
		--> Writing --> Volume Imperfections
	---------------------------------------------------------------------------------------
	--> Writing -->  2. Questions on Mechanical Properties 
	---------------------------------------------------------------------------------------
		--> Writing --> Tensile Strength
		--> Writing --> Yield Strength
		--> Writing --> Compressive Strength
		--> Writing --> Ductility
		--> Writing --> Malleability
		--> Writing --> Stiffness
		--> Writing --> Toughness
		--> Writing --> Creep Strength
		--> Writing --> Hardness
		--> Writing --> Impact Strength
		--> Writing --> Fatigue Strength
		--> Writing --> Formability
		--> Writing --> Structure Mechanical Property Relation
	---------------------------------------------------------------------------------------
	--> Writing -->  3. Questions & Answers on Dislocation Theory 
	------------------

		--> Writing --> Unit Dislocations and Partial Dislocations
		--> Writing --> Peierls-Nabarro Stress and Dislocation Width
		--> Writing --> Dislocation Multiplication
		--> Writing --> Dislocation Intersection
		--> Writing --> Dislocations in Ceramics
	---------------------------------------------------------------------------------------
	--> Writing -->  4. Questions on Deformation of Metals  
	---------------------------------------------------------------------------------------
		--> Writing --> Elastic Deformation
		--> Writing --> Plastic Deformation
		--> Writing --> Deformation by Twinning
		--> Writing --> Deformation Behaviour in Ceramics and Polymers
	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Strengthening Mechanisms in Materials 
	---------------------------------------------------------------------------------------
		--> Writing --> Grain Boundary Strengthening
		--> Writing --> 

	---------------------------------------------------------------------------------------
	--> Writing -->  6. Questions on Screening Libraries 
	---------------------------------------------------------------------------------------
		--> Writing --> Database Screening by Nucleic Acid Hybridisation-1
		--> Writing --> Database Screening by Nucleic Acid Hybridisation-2
		--> Writing --> Screening by Expression In Vivo
		--> Writing --> Expression of Coding Section in-Vitro
		--> Writing --> Reporter Genes
	---------------------------------------------------------------------------------------
	--> Writing -->  7. Questions & Answers on Modifications and Mutagenesis 
	---------------------------------------------------------------------------------------
		--> Writing --> Generation of Insertions and Deletions
		--> Writing --> Mutagenesis without PCR
		--> Writing --> Recovery of Mutated Sequences
		--> Writing --> Database of Mutant Lines and Gene Disruption
		--> Writing -->  Post-tra

		--> Writing --> Coordinated Regulation of Glycolysis
		--> Writing --> Coordinated Regulation of Glycogen Synthesis and Breakdown
		--> Writing --> Analysis of Metabolic Control
	---------------------------------------------------------------------------------------
	--> Writing -->  11. Questions & Answers on Amino Acid Oxidation and the Production of Urea 
	---------------------------------------------------------------------------------------
		--> Writing --> Metabolic Fates of Amino Groups
		--> Writing --> Nitrogen Excretion and the Urea Cycle
		--> Writing --> Pathways of Amino Acid Degradation
	---------------------------------------------------------------------------------------
	--> Writing -->  12. Questions on Oxidative Phosphorylation and Photophosphorylation 
	---------------------------------------------------------------------------------------
		--> Writing --> Electron-Transfer reactions in Mitochondria
		--> Writing --> ATP Synthesis-1
		--> Writing --> ATP Synthe

		--> Writing --> RNA Secondary Structure Prediction Methods
		--> Writing --> Ab Initio Approach
		--> Writing --> Comparative Approach
		--> Writing --> Performance Evaluation
		--> Writing --> Limitations of Prediction
		--> Writing --> Minimum Free – Energy Method & Stochastic Context – Free Grammars
		--> Writing --> MFOLD and the Use of Energy Plots
		--> Writing --> Searching Genomes for RNA & RNA Structure Modeling Applications
	---------------------------------------------------------------------------------------
	--> Writing -->  9. Questions & Answers on Genome Mapping, Assembly and Comparison 
	---------------------------------------------------------------------------------------
		--> Writing --> Genome Mapping
		--> Writing --> Genome Sequencing
		--> Writing --> Genome Sequence Assembly
		--> Writing --> Genome Annotation
		--> Writing --> Comparative Genomics
	---------------------------------------------------------------------------------------
	--> Writing -->  10.

		--> Writing --> Energy Balance Numericals Without Reactions
		--> Writing --> Enthalpy Change Due to Reaction
		--> Writing --> Heat of Reaction for Processes with Biomass Production
	---------------------------------------------------------------------------------------
	--> Writing -->  6. Questions on Unsteady – State Material and Energy Balances 
	---------------------------------------------------------------------------------------
		--> Writing --> Solving Unsteady – State Mass Balances
		--> Writing --> Solving Unsteady – State Energy Balances
	---------------------------------------------------------------------------------------
	--> Writing -->  7. Questions & Answers on Fluid Flow and Mixing  
	---------------------------------------------------------------------------------------
		--> Writing --> Classification of Fluids 
		--> Writing --> Fluids in Motion
		--> Writing --> Non Newtonian Fluids
		--> Writing --> Viscosity Measurement 
		--> Writing --> Factors Affecting

		--> Writing --> DNA Topology – 3
		--> Writing --> Chromatin Centromere and Telomere
	---------------------------------------------------------------------------------------
	--> Writing -->  3. Questions & Answers on Manipulation of Nucleic Acids & Polymerase Chain Reaction 
	---------------------------------------------------------------------------------------
		--> Writing --> Manipulating DNA
		--> Writing --> Hybridization of DNA and RNA
		--> Writing --> Fundamentals of PCR
		--> Writing --> Inverse PCR, Randomly Amplified Polymorphic DNA(RAPD)
		--> Writing --> Reverse Transcriptase PCR
	---------------------------------------------------------------------------------------
	--> Writing -->  4. Questions on DNA Sequencing 
	---------------------------------------------------------------------------------------
		--> Writing --> General Principles for Chain Termination Sequencing
		--> Writing --> Automated Sequencing
	----------------------------------------------------------

		--> Writing --> Microbial  Classification
	---------------------------------------------------------------------------------------
	--> Writing -->  2. Questions on The Microscopic Examination of Microorganisms 
	---------------------------------------------------------------------------------------
		--> Writing --> Light Microscope
		--> Writing --> Transmission Electron Microscope and Scanning Electron Microscope
		--> Writing --> Microbial Stains
	---------------------------------------------------------------------------------------
	--> Writing -->  3. Questions & Answers on The Morphology and Fine Structure of Bacteria 
	---------------------------------------------------------------------------------------
		--> Writing --> The Size, Shape and Arrangement of Bacterial Cells
		--> Writing --> Structures External to the Cell Wall
		--> Writing --> The Cell Wall
		--> Writing --> Structures Internal to the Cell Wall
	--------------------------------------------------------------

		--> Writing --> Industrial Uses of Bacteria
		--> Writing --> Industrial Uses of Yeasts
		--> Writing --> Industrial Uses of Molds
		--> Writing --> Genotypic Changes
		--> Writing --> Bacterial Recombination
		--> Writing --> Bacterial Conjugation
		--> Writing --> Bacterial Transduction
		--> Writing --> Bacterial Transformation
	---------------------------------------------------------------------------------------
	--> Writing -->  19. Questions & Answers on Microbial Flora of the Healthy Human Host, Host-Microbe Interactions : The Process of Infection 
	---------------------------------------------------------------------------------------
		--> Writing --> Normal Flora and the Human Host
File Not Found Error Handled
		--> Writing --> Distribution and Occurrence of the Normal Flora
File Not Found Error Handled
		--> Writing --> Microbial Adherence
		--> Writing --> Penetration of Epithelial Cell Layers
File Not Found Error Handled
		--> Writing --> Events in Infection Following 

		--> Writing --> Other Methods of Sex Determination
		--> Writing --> Sex Determination in Mammals
		--> Writing --> Sex Linked Inheritance
		--> Writing --> Dosage Compensation Mechanism for X linked Genes
		--> Writing --> Sex Influenced Traits and Sex Limited Traits
	---------------------------------------------------------------------------------------
	--> Writing -->  11. Questions & Answers on Extension and Deviation from Mendelian Genetic Principles 
	---------------------------------------------------------------------------------------
		--> Writing --> Incomplete Dominance and Codominance
		--> Writing --> Multiple Alleles
		--> Writing --> Gene Interactions
		--> Writing --> Epistasis 
		--> Writing --> Duplicate Epistasis
		--> Writing --> Lethal Alleles
	---------------------------------------------------------------------------------------
	--> Writing -->  12. Questions on Regulating Gene Expression in Eukaryotes 
	------------------------------------------------------

		--> Writing --> Insertion and Replacement Vectors
	---------------------------------------------------------------------------------------
	--> Writing -->  4. Questions on Cloning Vectors for Eukaryotes 
	---------------------------------------------------------------------------------------
		--> Writing --> Vectors for Yeast – 1
		--> Writing --> Vectors for Yeast – 2
		--> Writing --> Vectors for Yeast – 3
		--> Writing --> Vectors for Yeast – 4
		--> Writing --> Vectors for Plants – 1
		--> Writing --> Vectors for Plants – 2
		--> Writing --> Vectors for Plants – 3
		--> Writing --> Vectors for Insects
		--> Writing --> Vectors for Mammals – 1
		--> Writing --> Vectors for Mammals – 2
		--> Writing --> Vectors for Mammals – 3
		--> Writing --> Vectors for Mammals – 4
		--> Writing --> Vectors for Mammals – 5
		--> Writing --> Vectors for Mammals – 6
		--> Writing --> Vectors for Mammals – 7
		--> Writing --> Vectors for Mammals – 8
		--> Writing --> Vectors for Mammals – 9
		-->

		--> Writing --> Cell Signaling
		--> Writing --> Cell Cycle
		--> Writing --> Cancer
		--> Writing --> Cell Division
	---------------------------------------------------------------------------------------
	--> Writing -->  4. Questions on Prokaryotes and Viruses 
	---------------------------------------------------------------------------------------
		--> Writing --> Bacterial Taxonomy
		--> Writing --> Bacterial Genome
		--> Writing --> Bacterial Nutrition
		--> Writing --> Classification System
	---------------------------------------------------------------------------------------
	--> Writing -->  5. Questions & Answers on Immunology 
	---------------------------------------------------------------------------------------
		--> Writing --> Immunology Introduction
		--> Writing --> Cells of Immune System
		--> Writing --> Major Histocompatibility Complex
		--> Writing --> Immunoglobulins
		--> Writing --> B-Cell
		--> Writing --> T-Cells
	----------------------------------------

-------------------------------------------------------------------------------------------
--> Subject Key --> 407
--> Writing --> 1000 Human Anatomy and Physiology MCQs
-------------------------------------------------------------------------------------------
	---------------------------------------------------------------------------------------
	--> Writing -->  1. Questions & Answers on Osteology & Syndesmology
	---------------------------------------------------------------------------------------
		--> Writing --> Bone & Vertebral Column
		--> Writing --> Thorax & Skull
		--> Writing --> Bones of the Upper & Lower Extremity
		--> Writing --> Foot & Bones of the Lower Extremity
		--> Writing --> Development & Classification of Joints
		--> Writing --> Articulations of Trunk, Upper & Lower Extremity
	---------------------------------------------------------------------------------------
	--> Writing -->  2. Questions on Angiology, Arteries & Veins
	-------------------------------

		--> Writing --> Mechanisms of Drug Absorption
		--> Writing --> Factors Influencing Drug Absorption and Bioavailability
		--> Writing --> Physicochemical Factors Affecting Drug Absorption
		--> Writing --> Factors Affecting Rate of Drug Dissolution
		--> Writing --> pH Partition Hypothesis
		--> Writing --> Limitations of pH Partition Hypothesis
		--> Writing --> Dosage Form Factors Affecting Drug Absorption – 1
		--> Writing --> Dosage Form Factors Affecting Drug Absorption – 2
		--> Writing --> Nature and Type of Dosage Forms
		--> Writing --> Absorption of Drugs from Non Oral Extravascular Sites – 1
		--> Writing --> Absorption of Drugs from Non Oral Extravascular Sites – 2
	---------------------------------------------------------------------------------------
	--> Writing -->  2. Questions on Distribution of Drugs 
	---------------------------------------------------------------------------------------
		--> Writing --> Physiological Barriers to Drug Distribution
		--> Writing -

-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
-------------------------------------------------------------------------------------------
--> Subject Key --> 411
--> Writing --> 1000 Cell Biology MCQs
-------------------------------------------------------------------------------------------
	---------------------------------------------------------------------------------------
	--> Writing -->  1. Questions & Answers on Introduction to Cell
	---------------------------------------------------------------------------------------
		--> Writing --> Discovery of Cells
		--> Writing --> Basic Properties of Cells
		--> Writing --> Two Fundamentally Different Classes of Cells – 1
		--> Writing --> Two Fundamentally Different Classes of Cells – 2
		--> Writing --> Viruses and Viroids
	------------------------------------------------------------------------

		--> Writing --> G-protein Receptors and Messengers
		--> Writing --> Protein-tyrosine Phosphorylation
		--> Writing --> Role of Calcium
		--> Writing --> Crosstalk among Pathways
		--> Writing --> Role of NO
		--> Writing --> Apoptosis
	---------------------------------------------------------------------------------------
	--> Writing -->  17. Questions & Answers on Cancer 
	---------------------------------------------------------------------------------------
		--> Writing --> Properties of Cancer Cells
		--> Writing --> Causes of Cancer
		--> Writing --> Genetics of Cancer
		--> Writing --> Combating Cancer
	---------------------------------------------------------------------------------------
	--> Writing -->  18. Questions on the Immune Response 
	---------------------------------------------------------------------------------------
		--> Writing --> The Immune Response
		--> Writing --> Clonal Selection Theory
		--> Writing --> T-lymphocytes
		--> Writing --> Molecular Basis

		--> Writing --> Electron Spectroscopy for Chemical Analysis
		--> Writing --> Auger Electron Spectroscopy
		--> Writing --> Instrumentation for Electron Spectroscopy
		--> Writing --> Ion Spectroscopy
		--> Writing --> Instrumentation of Ion Spectroscopy
	---------------------------------------------------------------------------------------
	--> Writing -->  12. Questions on Electron Spin Resonance Spectrometers 
	---------------------------------------------------------------------------------------
		--> Writing --> Electron Spin Resonance
		--> Writing --> Basic ESR Spectrometer
		--> Writing --> Instrumentation of ESR Spectrometer
	---------------------------------------------------------------------------------------
	--> Writing -->  13. Questions & Answers on Digital Circuits
	---------------------------------------------------------------------------------------
		--> Writing --> Digital Circuits
		--> Writing --> Logic Families
----------------------------------------------

In [3]:
import requests
import os
from bs4 import BeautifulSoup
import re
req=requests.get('https://www.sanfoundry.com/')
soup=BeautifulSoup(req.text)
all_links=soup.find('div',class_='entry-content').find_all('a')
d={}
for i in all_links:
    d[i.text]=i.get('href')
x=list(d.keys())
y={}
j=0
for i,subject in enumerate(x):
    if re.match('1000.*?MCQs',subject):
        j+=1
        name=str(j)+". "+"_".join(subject.split()[1:-1])
        y[i]=name

In [4]:
y

{5: '1. Data_Structures_&_Algorithms_I',
 6: '2. Data_Structures_&_Algorithms_II',
 7: '3. Python',
 8: '4. Java',
 9: '5. C++',
 10: '6. C',
 11: '7. C#',
 39: '8. Object_Oriented_Programming',
 41: '9. Linux',
 44: '10. JUnit',
 45: '11. JavaScript',
 47: '12. Spring',
 49: '13. SAN',
 50: '14. PHP',
 53: '15. Ruby',
 54: '16. Hadoop',
 56: '17. R_Programming',
 58: '18. Cloud_Computing',
 60: '19. Data_Science',
 62: '20. MongoDB',
 64: '21. SQL_Server',
 66: '22. MySQL',
 68: '23. Oracle',
 69: '24. HTML',
 70: '25. CSS',
 71: '26. IOT',
 73: '27. Unix',
 75: '28. Computer_Fundamentals',
 77: '29. Cyber_Security',
 79: '30. Operating_System',
 81: '31. Computer_Architecture',
 83: '32. Software_Engineering',
 85: '33. Software_Architecture',
 87: '34. Artificial_Intelligence',
 89: '35. LISP_Programming',
 90: '36. Computer_Network',
 92: '37. Cryptograph_&_Network_Security',
 93: '38. Microprocessor',
 95: '39. Microcontroller',
 97: '40. Computer_Graphics',
 98: '41. Database_Man

In [11]:
!ls

Assignments
Material
Practice
SF
SF-Copy1.ipynb
SF.ipynb
Whatsapp Notebooks


In [12]:
!cd SF

In [19]:
!cat "SF/251251. Machine_Drawing/1._Questions_&_Answers_on_Drawing_Fundamentals/Questions/Method_of_Expression.txt

----Project By Karan Agrawal & Chirag Agarwal----



1. Axonometric view is a type of _____________
a) Orthographic view
b) Pictorial view
c) Oblique View
d) Pictorial and Oblique View


2. In an axonometric view, line of site are   ______________
a) Perpendicular to plane of projection with all three faces equally inclined to picture plane
b) Perpendicular to plane of projection with all three faces need not be equally inclined to picture plane
c) Inclined to the plane of projection at certain angle
d) One face of the object needs to be parallel


3. The figure shown below depicts which type of view?

a) Oblique View
b) Perspective View
c) Axonometric View
d) Orthographic View

https://www.sanfoundry.com/wp-content/uploads/2017/11/machine-drawing-questions-answers-method-expression-q3-300x229.png

4. Which type of view is shown in the below figure?

a) Oblique View
b) Perspective View
c) Axonometric View
d) Orthographic View

https://www.sanfoundry.com/wp-content/uploads/2017/11/machi

     5: '1. Data_Structures_&_Algorithms_I',
     6: '2. Data_Structures_&_Algorithms_II',
     7: '3. Python',
     8: '4. Java',
     9: '5. C++',
     10: '6. C',
     11: '7. C#',
     39: '8. Object_Oriented_Programming',
     41: '9. Linux',
     44: '10. JUnit',
     45: '11. JavaScript',
     47: '12. Spring',
     49: '13. SAN',
     50: '14. PHP',
     53: '15. Ruby',
     54: '16. Hadoop',
     56: '17. R_Programming',
     58: '18. Cloud_Computing',
     60: '19. Data_Science',
     62: '20. MongoDB',
     64: '21. SQL_Server',
     66: '22. MySQL',
     68: '23. Oracle',
     69: '24. HTML',
     70: '25. CSS',
     71: '26. IOT',
     73: '27. Unix',
     75: '28. Computer_Fundamentals',
     77: '29. Cyber_Security',
     79: '30. Operating_System',
     81: '31. Computer_Architecture',
     83: '32. Software_Engineering',
     85: '33. Software_Architecture',
     87: '34. Artificial_Intelligence',
     89: '35. LISP_Programming',
     90: '36. Computer_Network',
     92: '37. Cryptograph_&_Network_Security',
     93: '38. Microprocessor',
     95: '39. Microcontroller',
     97: '40. Computer_Graphics',
     98: '41. Database_Management',
     100: '42. RDBMS',
     101: '43. Discrete_Mathematics',
     103: '44. Compiler',
     104: '45. Embedded_Systems',
     106: '46. Automata_Theory',
     108: '47. Engineering_Mathematics',
     109: '48. Probability_and_Statistics',
     110: '49. Numerical_Methods',
     111: '50. Engineering_Physics_I',
     113: '51. Engineering_Physics_II',
     115: '52. Engineering_Chemistry_I',
     117: '53. Engineering_Chemistry_II',
     119: '54. Engineering_Mechanics',
     121: '55. Engineering_Drawing',
     123: '56. Professional_Communication',
     125: '57. Basic_Electrical_Engineering',
     127: '58. Basic_Chemical_Engineering',
     129: '59. Basic_Civil_Engineering',
     131: '60. Applied_Chemistry',
     133: '61. Heat_Transfer',
     135: '62. Mass_Transfer',
     137: '63. Heat_Transfer_Operations',
     139: '64. Polymer_Engineering',
     140: '65. Unit_Processes',
     141: '66. Separation_Processes',
     142: '67. Mechanical_Operations',
     143: '68. Distillation_Design',
     144: '69. Pulp_&_Paper',
     145: '70. Food_Engineering',
     147: '71. Thermodynamics',
     148: '72. Chemical_Process_Calculation',
     149: '73. Organic_Chemistry',
     151: '74. Materials_Science',
     153: '75. Fluidization_Engineering',
     154: '76. Civil_Engineering_Drawing',
     155: '77. Fermentation_Technology',
     157: '78. Computational_Fluid_Dynamics',
     197: '79. Electric_Circuits',
     198: '80. Electric_Drives',
     200: '81. Digital_Signal_Processing',
     202: '82. Digital_Communications',
     204: '83. Linear_Integrated_Circuits',
     205: '84. Digital_Image_Processing',
     206: '85. Electrical_Machines',
     207: '86. Design_of_Electrical_Machines',
     208: '87. Electromagnetic_Theory',
     209: '88. Power_Systems',
     210: '89. Power_Electronics',
     211: '90. Network_Theory',
     212: '91. Signals_and_Systems',
     213: '92. VLSI',
     214: '93. Control_Systems',
     215: '94. Microwave_Engineering',
     216: '95. Digital_Circuits',
     217: '96. Analog_Circuits',
     218: '97. Electronic_Devices_&_Circuits',
     219: '98. Neural_Networks',
     220: '99. Optical_Communications',
     221: '100. Electrical_Measurements',
     222: '101. DC_Machines',
     223: '102. Transformers',
     224: '103. Analog_Communications',
     225: '104. Biomedical_Instrumentation',
     227: '105. Wireless/Mobile',
     228: '106. VHDL',
     230: '107. MATLAB',
     232: '108. Dairy_Engineering',
     234: '109. Strength_of_Materials',
     236: '110. Theory_of_Machines',
     237: '111. Irrigation_Engineering',
     239: '112. Avionics',
     241: '113. Aerodynamics',
     242: '114. Aircraft_Design',
     244: '115. Aircraft_Maintenance',
     246: '116. Rocket_Propulsion',
     247: '117. Aerospace_Materials_&_Processes',
     248: '118. Fluid_Mechanics',
     249: '119. Hydraulic_Machines',
     250: '120. Machine_Design',
     251: '121. Machine_Drawing',
     252: '122. Machine_Dynamics',
     253: '123. Machine_Kinematics',
     254: '124. Manufacturing_Engineering_I',
     256: '125. Manufacturing_Engineering_II',
     258: '126. GeoTechnical_Engineering',
     260: '127. Engineering_Geology',
     262: '128. Environmental_Engineering',
     264: '129. Highway_Engineering',
     266: '130. Structural_Analysis',
     267: '131. Construction_&_Building_Materials',
     269: '132. Concrete_Technology',
     271: '133. Prestressed_Concrete_Structures',
     273: '134. Design_of_Steel_Structures',
     274: '135. Hazardous_Waste_Management',
     276: '136. Waste_Water_Engineering',
     278: '137. Surveying',
     362: '138. Casting-Forming-Welding_I',
     364: '139. Casting-Forming-Welding_II',
     366: '140. Machine_Tools',
     367: '141. Advanced_Machining',
     368: '142. IC_Engine',
     369: '143. Power_Plant_Engineering',
     370: '144. Steam_and_Gas_Turbines',
     371: '145. Automobile_Engineering',
     372: '146. Energy_Engineering',
     374: '147. Energy_Management',
     376: '148. Statistical_Quality_Control',
     378: '149. Engineering_Metrology',
     380: '150. Foundry_and_Forging',
     381: '151. Engineering_Material_&_Metallurgy',
     383: '152. Mechanical_Behaviour',
     385: '153. Testing_of_Materials',
     387: '154. Genetic_Engineering',
     389: '155. Biochemistry',
     391: '156. Bioinformatics',
     392: '157. Bioprocess_Engineering',
     393: '158. Molecular_Biology',
     395: '159. Microbiology',
     397: '160. Cytogenetics',
     399: '161. Clinical_Science',
     401: '162. Vector_Biology',
     403: '163. Life_Sciences',
     405: '164. Solid_State_Chemistry',
     406: '165. Human_Anatomy_and_Physiology',
     408: '166. Drug_Biotechnology',
     410: '167. Cell_Biology',
     412: '168. Analytical_Instrumentation',
     414: '169. Instrumentation_Transducers'